### PR TITLE
Add script to verify 1004 issue

### PR DESCRIPTION
**Pull request by HSavynetska**
ATF Test Scripts to check #[1004](https://github.com/smartdevicelink/sdl_core/issues/1004)

This PR is **not ready** for review.

### Summary
A test script has been added to verify that the problem has been fixed.

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+-- In case:
+-- 1) Press "Audio Service - Show" button
+-- 2) Press "Start Service"
+-- 3) Press "Cancel" on pop-up
+-- 4) Press "Cancel" on pop-up
+-- 5) Press "OK" pop-up
+-- 6) Press "Start File Streaming"
+-- Expected result:
+-- 1) Audio Streaming is started.
+-- Actual result:
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua]: Actual result is redundant, please remove

File: test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    local function responseSuccess()
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end
+    if exp.occurences >= 1 and exp.occurences <= 3 then
+      RUN_AFTER(responseRejected, delay)
+    elseif exp.occurences == 4 then
+      RUN_AFTER(responseSuccess, delay)
+    end
+  end)
+  :Times(AnyNumber())
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua]: Please expect the exact number of massages.

File: test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, init HMI, connect Mobile", common.start)
+runner.Step("Register App", common.registerApp)
+runner.Step("Activate App", common.activateApp)
+
+runner.Title("Test")
+runner.Step("StartStreaming", startStreaming)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua]: start streaming after the service start.

Comment (HSavynetska) [reply test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua]: @GetmanetsIrina You can find the updates in the commit 05f7cf9

File: test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua (hunk 2 above)
Comment (HSavynetska) [reply test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua]: @GetmanetsIrina You can find the updates in the commit 05f7cf9

File: test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua (hunk 1 above)
Comment (HSavynetska) [reply test_scripts/Defects/5_1/1004_App_is_disconnected_due_to_PROTOCOL_VIOLATION_when_start_audio_streaming_after_rejected_2_times_then_accepted.lua]: @GetmanetsIrina You can find the updates in the commit 05f7cf9

**Comment by dboltovskyi** [comment]
Not actual. Sequence is covered in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2469

**Pull request closed without merging**